sys/cddl/contrib/opensolaris/uts/common/fs/zfs/zfs_znode.c

```diff
@@ -145,8 +145,9 @@ zfs_znode_cache_destructor(void *buf, void *arg)
 	znode_t *zp = buf;
 
 	ASSERT(!POINTER_IS_VALID(zp->z_zfsvfs));
-	ASSERT(ZTOV(zp) == NULL);
+	ASSERT(ZTOV(zp) != NULL);
```

Comment:
Yeah, the code looks weird because the assertion is FreeBSD specific where we can free a znode only after it is fully disconnected from a vnode and the `vn_free` call is from illumos where the filesystem driver has to explicitly return the vnode to the system (as opposed to FreeBSD where the system tells I want to reclaim this vnode and the driver has to obey).